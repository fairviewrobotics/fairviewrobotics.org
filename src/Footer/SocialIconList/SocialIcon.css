.social-icon {
  height: 1.5em;
  width: 3em;
  cursor: pointer;
  background-size: auto 100%;
  background: no-repeat center center;
  opacity: 1;
  transition: color 0.25s ease;
  color: var(--main-color);
  font-size: 1.65em;
  text-align: center;
}

.social-icon:hover {
  color: var(--second-color);
}