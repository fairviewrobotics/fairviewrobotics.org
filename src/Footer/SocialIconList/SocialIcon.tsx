import React, { PureComponent } from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

import styles from './SocialIcon.module.css';

export type SocialIconProps = {
  icon: any;
  url: string;
}

export default class SocialIcon extends PureComponent<SocialIconProps> {
  render() {
    return (
      <a 
        className={styles.socialIcon}
        target="_blank"
        rel="noopener noreferrer"
        href={this.props.url}>
        <FontAwesomeIcon icon={this.props.icon} />
      </a>
    );
  }
}
