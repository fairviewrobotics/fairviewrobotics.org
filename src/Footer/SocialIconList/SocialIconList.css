.social-media-list {
  display: flex;
  position: absolute;
  height: 10vh;
  width: 30vw;
  align-items: center;
  justify-content: space-around;
  z-index: 10;
}

@media(max-width:768px) {
  .social-media-list {
    width: 80vw;
  }
}