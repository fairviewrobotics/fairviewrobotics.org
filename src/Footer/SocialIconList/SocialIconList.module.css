.socialMediaList {
  display: flex;
  position: absolute;
  height: 10vh;
  width: 30vw;
  align-items: center;
  justify-content: space-around;
  z-index: 10;
}

@media(max-width:768px) {
  .socialMediaList {
    width: 80vw;
  }
}