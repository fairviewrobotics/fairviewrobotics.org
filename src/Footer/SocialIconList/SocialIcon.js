import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import FontAwesomeIcon from '@fortawesome/react-fontawesome';

import './SocialIcon.css';

export default class SocialIcon extends PureComponent {

  static propTypes = {
    icon: PropTypes.any.isRequired,
    url: PropTypes.string.isRequired
  };

  render() {
    return (
      <a className="social-icon"
         target="_blank"
         href={this.props.url}>
        <FontAwesomeIcon icon={this.props.icon} />
      </a>
    );
  }
}
