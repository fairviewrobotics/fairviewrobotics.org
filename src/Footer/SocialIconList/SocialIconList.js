import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import SocialIcon from "./SocialIcon";

import './SocialIconList.css';

export default class SocialIconList extends PureComponent {

  static propTypes = {
    socialMedia: PropTypes.arrayOf(PropTypes.shape({
      url: PropTypes.string.isRequired,
      icon: PropTypes.object.isRequired
    })).isRequired
  };

  render() {
    return (
      <div className="social-media-list">
        {this.props.socialMedia.map(icon =>
          <SocialIcon key={icon.url} icon={icon.icon} url={icon.url} />
        )}
      </div>
    );
  }
}
