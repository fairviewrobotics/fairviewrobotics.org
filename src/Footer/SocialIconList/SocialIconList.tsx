import React, { PureComponent } from 'react';
import SocialIcon from "./SocialIcon";

import styles from './SocialIconList.module.css';

export type SocialIconListProps = {
  socialMedia: { url: string, icon: any }[];
}

export default class SocialIconList extends PureComponent<SocialIconListProps> {
  render() {
    return (
      <div className={styles.socialMediaList}>
        {this.props.socialMedia.map(icon =>
          <SocialIcon key={icon.url} icon={icon.icon} url={icon.url} />
        )}
      </div>
    );
  }
}
