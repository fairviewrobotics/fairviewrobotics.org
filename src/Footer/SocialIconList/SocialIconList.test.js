import React from "react";
import { shallow } from "enzyme";
import SocialIconList from "./SocialIconList";

describe('SocialIconList', () => {
  let wrapper;

  it('render without crashing', () => {
    const socialMedia = [];
    shallow(
      <SocialIconList socialMedia={socialMedia}/>
    );
  });

  describe('given a list of `socialMedia`', () => {
    beforeEach(() => {
      const socialMedia = [
        {
          icon: {
            name: 'githubIcon'
          },
          url: 'http://github.com'
        },
        {
          icon: {
            name: 'facebookIcon'
          },
          url: 'http://facebook.com'
        },
        {
          icon: {
            name: 'twitterIcon'
          },
          url: 'http://twitter.com'
        }
      ];
      wrapper = shallow(
        <SocialIconList socialMedia={socialMedia}/>
      );
    });

    it('should render a list of social media icons', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});