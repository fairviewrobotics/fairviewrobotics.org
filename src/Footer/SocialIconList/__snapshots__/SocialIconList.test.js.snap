// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialIconList given a list of \`socialMedia\` should render a list of social media icons 1`] = `
<div
  className="socialMediaList"
>
  <SocialIcon
    icon={
      Object {
        "name": "githubIcon",
      }
    }
    key="http://github.com"
    url="http://github.com"
  />
  <SocialIcon
    icon={
      Object {
        "name": "facebookIcon",
      }
    }
    key="http://facebook.com"
    url="http://facebook.com"
  />
  <SocialIcon
    icon={
      Object {
        "name": "twitterIcon",
      }
    }
    key="http://twitter.com"
    url="http://twitter.com"
  />
</div>
`;
